File: ObjectAvoidance/java/src/main/java/elite/gils/country/KH.java
package elite.gils.country;

/**
 * Continent: (Name)
 * Country: (Name)
*/

public enum KH {
    //List of State/Province names (Use 2 Letter/Number Combination codes)
    
    /**
     * Name:
    */
    AA("Example"),
    
    /**
     * Name:
    */
    AB("Example"),
    
    /**
     * Name:
    */
    AC("Example");
    
    //Leave the code below alone (unless optimizing)
    
    private final String name;
    
    KH(String name){
        this.name = name;
    }
    
    public String getName(KH state){
        return state.name;
    }
}

File: ObjectAvoidance/kotlin/src/elite/gils/country/KH.kt
package elite.gils.country

/**
 * Continent: (Name)
 * Country: (Name)
 */

enum class KH private constructor(
        //Leave the code below alone (unless optimizing)

        private val name: String) {
    //List of State/Province names (Use 2 Letter/Number Combination codes)

    /**
     * Name:
     */
    AA("Example"),

    /**
     * Name:
     */
    AB("Example"),

    /**
     * Name:
     */
    AC("Example");

    fun getName(state: KH): String {
        return state.name
    }
}